### PR TITLE
Remove xbuild dependency.

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Pull down nuget and msbuild in the build.sh script. Check for cURL and fail out if missing.

@weshaggard, @akoeplinger, @chcosta, @ellismg

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -19,8 +24,30 @@ fi
 __buildproj=$__scriptpath/build.proj
 __buildlog=$__scriptpath/msbuild.log
 
-# Run the restore build tools under xbuild to get NuGet and MSBuild pulled down
-xbuild "$__buildproj" /t:_RestoreBuildTools /nologo /verbosity:minimal "/fileloggerparameters:Verbosity=diag;LogFile=$__buildlog"
+# Pull NuGet.exe down if we don't have it already
+if [ ! -e "$__nugetpath" ]; then
+    which curl > /dev/null 2> /dev/null
+    if [ $? -ne 0 ]; then
```
Comment (akoeplinger) [inline build.sh]: We should check for wget as well, it's very likely that at least one of the two is installed on almost all systems.

File: build.sh [hunk 2]
```diff
@@ -19,8 +24,30 @@ fi
...
+if [ ! -e "$__msbuildpath" ]; then
+    echo "Restoring MSBuild..."
+    mono "$__nugetpath" install $__msbuildpackageid -Version $__msbuildpackageversion -ConfigFile "$__nugetconfig" -OutputDirectory "$__packageroot"
+    if [ $? -ne 0 ]; then
+        echo "Failed to restore MSBuild."
+        exit 1
+    fi
+fi
 
 if [ ! -e "$__msbuildpath" ]; then
```
Comment (akoeplinger) [inline build.sh]: Do we still need this check? If something goes wrong during nuget install of MSBuild in the previous step we should get an error anyway.

File: build.sh (hunk 1 above)
Comment (JeremyKuhne) [inline build.sh]: Any reason to use cURL over wget? wget seems to be installed by default on Ubuntu?

File: build.sh (hunk 2 above)
Comment (JeremyKuhne) [inline build.sh]: It is pretty unlikely to get hit, true. I can remove it if you think it is confusing or generally a net negative.

File: build.sh (hunk 1 above)
Comment (akoeplinger) [inline build.sh]: wget doesn't ship with OSX, but curl does. We need to check for both :)

File: build.sh (hunk 2 above)
Comment (akoeplinger) [inline build.sh]: I think it complicates the script for little value, so I'd prefer to just remove it.

File: build.sh (hunk 1 above)
Comment (JeremyKuhne) [inline build.sh]: Good to know, thanks. Need to get a Mac. :)

File: build.sh (hunk 2 above)
Comment (JeremyKuhne) [inline build.sh]: Fair enough, removing.

**Comment by JeremyKuhne** [comment]
Updated to look for cURL or wget.

**Comment by akoeplinger** [comment]
LGTM, thanks :+1:

**Comment by ellismg** [comment]
Nice!  LGTM.

**Comment by JeremyKuhne** [comment]
@dotnet-bot test this please

**Pull request merged**